pkg/controller/core/workload_controller.go

```diff
@@ -750,7 +750,7 @@ func (w *workloadCqHandler) Update(ctx context.Context, ev event.UpdateEvent, wq
 
 	if !newCq.DeletionTimestamp.IsZero() ||
 		!slices.CmpNoOrder(oldCq.Spec.AdmissionChecks, newCq.Spec.AdmissionChecks) ||
-		oldCq.Spec.StopPolicy != newCq.Spec.StopPolicy {
+		ptr.Equal(oldCq.Spec.StopPolicy, newCq.Spec.StopPolicy) {
```

Comment:
```suggestion
		!ptr.Equal(oldCq.Spec.StopPolicy, newCq.Spec.StopPolicy) {
```

Sorry for this confusion. I meant this.

Reply:
Given the impact, and that it is hard to detect, maybe we could have a unit test for it, potentially in a follow up?